openlcsd/flow/tasks.py

```diff
@@ -293,23 +293,62 @@ def download_shared_remote_source(context, engine):
     for shared use when corgi download_url invalid
     """
     parent_component = context.get('parent_component')
-    if not parent_component:
-        raise RuntimeError("no parent_component info when "
-                           "try to download shared remote source")
-
     config = context.get('config')
     koji_connector = KojiConnector(config)
     sc = SourceContainerHandler(config=config)
 
     # mark this component use shared remote source
     context['shared_remote_source'] = True
 
-    rs_root_dir = sc.get_shared_remote_source_root_dir(
-        context.get('tmp_root_dir'), parent_component)
+    if not parent_component:
+        # no parent component info, get from corgi
+        component = context.get('component')
+        connector = CorgiConnector()
+        component = connector.get(component.get("link"))
+        sources = component["sources"]
+        oci_ts = 0
+        # find the newest build oci according to build completion time
+        for source in sources:
+            if not source['purl'].startswith("pkg:oci"):
+                continue
+            # get build complete time from brew, get the newest one
+            oci_component = connector.get(
+                source['link'],
+                includes=connector.default_includes_type_independence
+            )
+            build_info = koji_connector.get_build(
+                int(oci_component["software_build"]['build_id']))
```

Comment:
this assumes `oci_component` returned from corgi always have non-empty software_build.

Reply:
Yes. Is there any chance oci component have empty software_build field.

Reply:
I didn't find one right now. but I think it is literally possible to have one with `namespace` other than `REDHAT`.

I think it's ok to keep it this way for now.